### PR TITLE
ci: migrate to GitHub Actions from Travis CI

### PR DESCRIPTION
**Pull request by komainu8**
I migrate regression tests from TravisCI to Github Actions.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Install dependency
+        run: |
+          curl --silent --location https://github.com/groonga/groonga/raw/master/data/travis/setup.sh | sh
+          sudo gem install test-unit
+          sudo apt -y -V install libreadline-dev
+        env:
+          GROONGA_MASTER: yes
+      - name: Install PostgreSQL
+        run: |
+          wget https://ftp.postgresql.org/pub/source/v${{ matrix.postgresql-version }}/postgresql-${{ matrix.postgresql-version }}.tar.bz2
```
Comment (kou) [inline .github/workflows/linux.yml]: Could you use the official package instead of building from source?

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        with:
+          fetch-depth: 1
+          submodules: true
+      - name: Install dependency
+        run: |
+          curl --silent --location https://github.com/groonga/groonga/raw/master/data/travis/setup.sh | sh
+          sudo gem install test-unit
+          sudo apt -y -V install libreadline-dev
+        env:
+          GROONGA_MASTER: yes
```
Comment (kou) [inline .github/workflows/linux.yml]: Could you stop using `GROONGA_MASTER: yes` on all jobs?

**Comment by kou** [comment]
All tests are failed:

```quote
psql: error: could not connect to server: FATAL:  role "root" does not exist
```

**Comment by komainu8** [comment]
@kou Tahnk you for your comments.
I've fixed them.

**Pull request merged**